src/internal/utils/date-time/__tests__/format-date-localized.test.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import formatDateLocalized from '../../../../../lib/components/internal/utils/date-time/format-date-localized';
+
+describe('formatDateLocalized', () => {
+  test('formats date correctly when the date is not ISO formatted', () => {
+    expect(
+      formatDateLocalized({
+        date: 'Tue Nov 12 2024 11:03:17 GMT+0100 (Central European Standard Time)',
+        isDateOnly: true,
+        locale: 'en-US',
+      })
+    ).toBe('November 12, 2024');
+  });
+
+  test('formats date correctly when the TZ is negative in relation to UTC', () => {
+    // Store original timezone offset method
+    const originalTimezoneOffset = Date.prototype.getTimezoneOffset;
+
+    // Mock to simulate Pacific Time (UTC-8/UTC-7)
```

Comment:
as an alternative, it could be achieved via using https://www.npmjs.com/package/timezone-mock lib